vulture_os/toolkit/api_parser/trendmicro_visionone/trendmicro_visionone.py

```diff
@@ -56,75 +59,120 @@ def __init__(self, data):
-        try:
-            while True:
                 logger.info(f"[{__parser__}]:__execute_query: URL: {link} , params: {query}", extra={'frontend': str(self.frontend)})
                 r = requests.get(link, params=query, headers=headers,proxies=self.proxies, verify=self.api_parser_verify_ssl, timeout=timeout)
                 if r.status_code != 200:
                     status = False
-                    logger.error(f"Error on URL: {link} Status: {r.status_code} Reason/Content: {r.content}")
+                    logger.warning(f"Error on URL: {link} Status: {r.status_code} Reason/Content: {r.content}")
                     raise TrendmicroVisionOneAPIError(f"Error on URL: {link} Status: {r.status_code} Reason/Content: {r.content}")
 
-                items.extend(r.json()['items'])
+                request_json = r.json()
+                if kind != "audit" and request_json.get('totalCount') >= MAX_TOTAL_COUNT:
+                    # Note : This attribute is not present in audit logs response
+                    raise requests.exceptions.ReadTimeout(f"Total count = {request_json.get('totalCount')} > {MAX_TOTAL_COUNT} (given max count)")
 
-                if link := r.json().get('nextLink'):
+                items.extend(request_json['items'])
+
+                if link := request_json.get('nextLink'):
                     query = None
-                    self.update_lock()
-                    continue
+                    has_next = True
                 else:
+                    has_next = False
+                    status = True
+            except requests.exceptions.ReadTimeout as e:
+                logger.exception(f"[{__parser__}]:__execute_query: {e}", extra={'frontend': str(self.frontend)})
+                has_next = False
+                to -= (to - since) / 2
+                if (to - since) < timedelta(seconds=MIN_QUERY_INTERVAL):
+                    logger.warning(
+                        f"[{__parser__}]:__execute_query: Failed on {kind} logs, time range too small between {since} and {to}",
+                        extra={'frontend': str(self.frontend)})
                     break
-        except Exception as e:
-            status = False
-            logger.error(f"[{__parser__}]:__execute_query: Error on link '{link}' : {e}", extra={'frontend': str(self.frontend)})
+                else:
+                    logger.info(
+                        f"[{__parser__}]:__execute_query: Will retry with smaller time range : {since} to {to}",
+                        extra={'frontend': str(self.frontend)})
 
-        return items, status
+            except Exception as e:
+                logger.warning(f"[{__parser__}]:execute: Failed on {kind} logs, between time of {since} and {to} : {e}",
+                             extra={'frontend': str(self.frontend)})
+                logger.exception(f"[{__parser__}]:__execute_query: {e}", extra={'frontend': str(self.frontend)})
+                retry -= 1
```

Comment:
Avoid catching `Exception` and at least handle those you expect could occur separately (for example `JSONDecodeError`, `HTTPError`, etc...)